awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsApplicationSignalsCustomizerProvider.java

```diff
@@ -121,6 +129,37 @@ static boolean isLambdaEnvironment() {
     return System.getenv(AWS_LAMBDA_FUNCTION_NAME_CONFIG) != null;
   }
 
+  static boolean isSigV4Enabled() {
+    String otlpEndpoint = System.getenv(OTEL_EXPORTER_OTLP_TRACES_ENDPOINT_CONFIG);
+    boolean isXrayOtlpEndpoint =
+        otlpEndpoint != null
+            && Pattern.compile(XRAY_OTLP_ENDPOINT_PATTERN)
+                .matcher(otlpEndpoint.toLowerCase())
+                .matches();
+
+    if (isXrayOtlpEndpoint) {
+      logger.log(Level.INFO, "Detected using AWS OTLP XRay Endpoint.");
+
+      String otlpTracesProtocol = System.getenv(OTEL_EXPORTER_OTLP_TRACES_PROTOCOL_CONFIG);
```

Comment:
Should this validation happen earlier before the regular expression matches? If it's not `http/protobuf`, we won't need to parse the endpoint and check.